mobile/src/pages/ProfilePage.tsx

```diff
@@ -1,37 +1,157 @@
-import { SafeAreaView, Image, View, Text, ScrollView } from "react-native";
+import { SafeAreaView, Image, View, Text, ScrollView, ActivityIndicator } from "react-native";
 import BottomNavBar from "../components/BottomNavBar";
 import TopNavBar from "../components/TopNavBar";
 import Button from "../components/Button";
-import Card from "../components/Card";
-import { useState } from "react";
+import { useEffect, useState } from "react";
+import { useQuery } from "@tanstack/react-query";
+import { getAuthSession } from "../utils/storage";
+import { getUser } from "../requests/fetch";
+import { ErrorResponse } from "../types/error.type";
+import useInfiniteQueryPosts from "../hooks/useInfiniteQueryPosts";
+import { Post } from "../types/post.type";
 
 export default ({ navigation }) => {
   const [search, setSearch] = useState("");
+  const [userId, setUserId] = useState(null);
+  const [userName, setUserName] = useState("");
+  const [userAvatar, setUserAvatar] = useState(null);
+
+  const [currentView, setCurrentView] = useState<"created" | "collections">(
+    "created"
+  );
+
+  useEffect(() => {
+    (async () => {
+      const { userId, userAvatar, userName } = await getAuthSession();
+      setUserId(userId);
+      setUserName(userName);
+      setUserAvatar(userAvatar);
+    })();
+  }, []);
+
+  const {
+    data: user,
+    isLoading: isProfileLoading,
+    refetch: refetchProfile,
+  } = useQuery({
+    // FIXME figure out how to incorporate userid here
+    queryKey: ["user_profile"],
+    queryFn: async () => {
+      const userId = (await getAuthSession()).userId;
+      return getUser({ userId });
```

Comment:
ohh btw, you dont actually need to get userId there. the getUser() should be getting it for you as long as you're logged in
L39 https://github.com/poop-team/daydream/blob/main/mobile/src/requests/fetch.ts
```
export async function getUser({ userId = "", userName = "" }) {
  const params = new URLSearchParams({
    userId: (await getAuthSession()).userId,
    searchUserId: userId || (await getAuthSession()).userId,
    userName: userName,
  });```

Reply:
Good to know, I will remove this